custom/icds_reports/utils/aggregation_helpers/birth_preparedness_forms.py

```diff
@@ -78,36 +80,38 @@ def aggregation_query(self):
-            ucr.anemia AS anemia,
-            ucr.eating_extra AS eating_extra,
-            ucr.resting AS resting,
-            ucr.anc_weight anc_weight,
-            ucr.anc_blood_pressure as anc_blood_pressure,
-            ucr.bp_sys as bp_sys,
-            ucr.bp_dia as bp_dia,
-            ucr.anc_hemoglobin as anc_hemoglobin,
-            ucr.bleeding as bleeding,
-            ucr.swelling as swelling,
-            ucr.blurred_vision as blurred_vision,
-            ucr.convulsions as convulsions,
-            ucr.rupture as rupture,
-            ucr.anc_abnormalities as anc_abnormalities,
+            COALESCE(ucr.anemia,prev_month.anemia) AS anemia,
+            COALESCE(ucr.eating_extra,prev_month.eating_extra) AS eating_extra,
+            COALESCE(ucr.resting,prev_month.resting) AS resting,
+            COALESCE(ucr.anc_weight,prev_month.anc_weight) anc_weight,
+            COALESCE(ucr.anc_blood_pressure,prev_month.anc_blood_pressure) as anc_blood_pressure,
+            COALESCE(ucr.bp_sys,prev_month.bp_sys) as bp_sys,
+            COALESCE(ucr.bp_dia,prev_month.bp_dia) as bp_dia,
+            COALESCE(ucr.anc_hemoglobin,prev_month.anc_hemoglobin) as anc_hemoglobin,
+            COALESCE(ucr.bleeding,prev_month.bleeding) as bleeding,
+            COALESCE(ucr.swelling,prev_month.swelling) as swelling,
+            COALESCE(ucr.blurred_vision,prev_month.blurred_vision) as blurred_vision,
+            COALESCE(ucr.convulsions,prev_month.convulsions) as convulsions,
+            COALESCE(ucr.rupture,prev_month.rupture) as rupture,
+            COALESCE(ucr.anc_abnormalities,prev_month.anc_abnormalities) as anc_abnormalities,
             COALESCE(ucr.valid_visits, 0) as valid_visits,
-            ucr.play_birth_preparedness_vid as play_birth_preparedness_vid,
-            ucr.counsel_preparation as counsel_preparation,
-            ucr.play_family_planning_vid as play_family_planning_vid,
-            ucr.conceive as conceive,
-            ucr.counsel_accessible_ppfp as counsel_accessible_ppfp
+            GREATEST(ucr.play_birth_preparedness_vid, prev_month.play_birth_preparedness_vid) as play_birth_preparedness_vid,
+            GREATEST(ucr.counsel_preparation, prev_month.counsel_preparation) as counsel_preparation,
+            GREATEST(ucr.play_family_planning_vid, prev_month.play_family_planning_vid) as play_family_planning_vid,
+            GREATEST(ucr.conceive,prev_month.conceive) as conceive,
+            GREATEST(ucr.counsel_accessible_ppfp, prev_month.counsel_accessible_ppfp) as counsel_accessible_ppfp,
+            COALESCE(ucr.ifa_last_seven_days, prev_month.ifa_last_seven_days) as ifa_last_seven_days,
+            COALESCE(ucr.using_ifa, prev_month.using_ifa) as using_ifa
           FROM ({ucr_table_query}) ucr
-          LEFT JOIN "{previous_month_tablename}" prev_month
+          FULL JOIN "{previous_month_tablename}" prev_month
```

Comment:
To make it satisfy the need of ever submitted/last submitted form. (Eg. if pregnant check if bp form submitted with immediate_breastfeeding='yes' prior to the end of month)

reference:  https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/utils/aggregation_helpers/postnatal_care_forms_ccs_record.py#L69